inference-engine/src/plaidml_plugin/ops/embedding_bag_offsets_sum.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace plaidml::edsl;
+
+namespace {
+
+template <typename T>
+std::vector<T> cast_constant_operand(size_t operand_idx, ngraph::Node* layer) {
+  auto* ngraph_const = ngraph::as_type<ngraph::op::Constant>(layer->get_input_node_ptr(operand_idx));
+  if (ngraph_const) {
+    return ngraph_const->cast_vector<T>();
+  } else {
+    THROW_IE_EXCEPTION << "Dynamic slicing not currently supported by PlaidML plugin; all of indices, and offsets"
+                          "must be Constants.";
+  }
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("EmbeddingBagOffsetsSum", [](const Context& ctx) {
+  auto* layer = ngraph::as_type<ngraph::opset4::EmbeddingBagOffsetsSum>(ctx.layer);
+  auto I = ctx.operands.at(0);
+  auto indices = ctx.operands.at(1);
+  auto indices_cst = cast_constant_operand<size_t>(1, layer);
+  auto offsets = cast_constant_operand<int32_t>(2, layer);
+  auto default_index = cast_constant_operand<int64_t>(3, layer);
+
```

Comment:
We should support a `per_sample_weights` input (i.e. input `4`). When it is not provided, assume it is the default value (a vector of 1s whose length is num_indices).